src/wallet/wallet.cpp

```diff
@@ -1262,6 +1269,15 @@ bool CWallet::AddToWalletIfInvolvingMe(const CTransactionRef& ptx, const CBlockI
                             LogPrintf("%s: Topping up keypool failed (locked wallet)\n", __func__);
                         }
                     }
+                    if (pIndex != nullptr && mapKeyMetadata[keyid].nCreateTime > pIndex->nTime) {
+                        LogPrintf("%s: Found a key which appears to be used earlier than we expected, updating metadata\n", __func__);
+                        CPubKey vchPubKey;
+                        bool res = GetPubKey(keyid, vchPubKey);
+                        assert(res); // this should never fail
```

Comment:
You don't use res really, would it be better to make this one line?

Reply:
We still need to call `GetPubKey` to get `vchPubKey` even when asserts are disabled. Technically, it shouldn't be possible to compile it that way (with `-DNDEBUG`) because of https://github.com/dashpay/dash/blob/master/src/validation.cpp#L61 but imo it's better to avoid introducing potential issues anyway.